core/src/main/scala/akka/persistence/r2dbc/journal/JournalDao.scala

```diff
@@ -190,6 +191,10 @@ private[r2dbc] class JournalDao(journalSettings: R2dbcSettings, connectionFactor
       row
     }
 
+    // Question: select with no limit may be problematic if we materialize to Future
+    // can't we have a selectStream variant returning a Source?
+    // I'm assuming that the driver won't fetch everything in one shot,
+    // but publish following the pace of the subscriber
```

Comment:
There are two reasons for materializing eagerly to Future[Seq].

1) If we would return a Source it would be difficult to know when the connection can be closed. That must be done when all rows have been mapped (consumed).

2) Connections is a limited resource and it's better to return it to the pool as quickly as possible rather than having it open while the potentially slow consumer is processing all rows.

Limit is used.